### PR TITLE
Feat: Adds `ButtonBuy` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Creates `ButtonBuy` component on @faststore/components
- Creates `ButtonBuy` styles on @faststore/ui
- Adds `ButtonBuy` documentation page on Docusaurus
- Updates `Button`  documentation page on Docusaurus

## How it works?

ButtonBuy is a custom Button that intends to send the user to the checkout/cart step directly.

## How to test it?

run yarn start in app/docs folder and checks the Icon component or test it using the doc [preview link.](https://faststore-git-feat-buy-button-fs-608-faststore.vercel.app/reference/ui/molecules/ButtonBuy)

### Starters Deploy Preview

Feat: Replaces ButtonBuy FSUI: https://github.com/vtex-sites/nextjs.store/pull/315

### CODE REVIEW

File: apps/docs/docs/reference/ui/molecules/ButtonBuy.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+---
+description: ButtonBuy intends to send the user to the checkout/cart step directly.
+keywords:
+  - UI library
+  - ecommerce
+  - react
+sidebar_custom_props:
+  image: https://vtexhelp.vtexassets.com/assets/docs/src/Button___6d6b16886d5d7e2a05dba8c45c075796.png
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/molecules/ButtonBuy.mdx:8]: @carolinamenezes we also need to update this preview image 🙈  thanks

File: apps/docs/docs/reference/ui/molecules/ButtonBuy.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  <TokenRow
+    token="--fs-button-buy-border-color-hover"
+    value="var(--fs-button-buy-bkg-color-hover)"
+    isColor
+  />
+  <TokenRow
+    token="--fs-button-buy-border-color-active"
+    value="var(--fs-button-buy-bkg-color-active)"
+    isColor
+  />
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/molecules/ButtonBuy.mdx:103]: Fan, try using the `GlobalValue` here in order to render the correct color.
image

**Review by renatamottam — APPROVED** [review]
Looks good to me!

**Review by eduardoformiga — APPROVED** [review]

**Comment by hellofanny** [comment]
> Shouldn't the component name be `BuyButton`?
> [We always call it that](https://developers.vtex.com/vtex-developer-docs/docs/vtex-store-components-buybutton)
> [Just like the marketplace (aka Shopify)](https://help.shopify.com/en/manual/online-sales-channels/buy-button)

hahaha. I was not sure, I named it BuyButton, then I changed to `ButtonBuy`.
@renatamottam  I will rename it again, ok?

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**